packages/cloudflare/src/cli/build/patches/plugins/fetch-cache-wait-until.ts

```diff
@@ -0,0 +1,42 @@
+import { patchCode } from "../ast/util.js";
+import type { ContentUpdater } from "./content-updater.js";
+
+/**
+ * The following Next.js code sets values in the incremental cache for fetch calls:
+ *  https://github.com/vercel/next.js/blob/e5fc495e3d4/packages/next/src/server/lib/patch-fetch.ts#L690-L728
+ *
+ * The issue here is that this promise is never awaited in the Next.js code (since in a standard node.js server
+ * the promise will eventually simply just run) but we do need to run it inside `waitUntil` (so that the worker
+ * is not killed before the promise is fully executed), without that this promise gets discarded and values
+ * don't get saved in the incremental cache.
+ *
+ * This function wraps the promise in a `waitUntil` call (retrieved from `globalThis.__openNextAls.getStore()`).
+ */
+export function patchFetchCacheSetMissingWaitUntil(updater: ContentUpdater) {
+  return updater.updateContent(
+    "patch-fetch-cache-set-missing-wait-until",
+    { filter: /\.(js|mjs|cjs|jsx|ts|tsx)$/ },
+    ({ contents }) => {
+      if (/Failed to set fetch cache/.test(contents)) {
+        return patchCode(contents, rule);
+      }
+    }
+  );
+}
+
+export const rule = `
+rule:
+  kind: return_statement
+  all:
+    - pattern: return $PROMISE, $CLONED2
+    - regex: Failed to set fetch cache
+    - follows:
+        kind: lexical_declaration
+        all:
+          - pattern: let [$CLONED1, $CLONED2]
+    
+
+fix: |
+  globalThis.__openNextAls.getStore().waitUntil($PROMISE);
```

Comment:
Have you considered using getCloudflareContext here? 
I don't really see pros nor cons, mostly curious

Reply:
I have but I have a slight preference to using the ALS directly since:
 - importing `getCloudflareContext` from our package which feels a bit unclean to me (the package that builds the code also adds an import to itself)
 - I feel that `getCloudflareContext` is slightly less stable than the ALS that just lives onto `globalThis` (for example we can change the `getCloudflareContext` api/behavior and subtly breaking this code)
 - this "lower level" access to the globalThis ALS seems appropriate in built code (while users should use `getCloudflareContext` for a higher level of abstraction)

But the above are not huge concerns if you preferred I would not particularly mind calling `getCloudflareContext`, I don't think it makes a huge difference

Reply:
I was curious but `__openNextAls` SGTM.

Thanks for sharing your reflexion.